src/segment-loader.js

```diff
@@ -51,20 +51,18 @@ const detectEndOfStream = function(playlist, mediaSource, segmentIndex) {
  * @extends videojs.EventTarget
  */
 export default class SegmentLoader extends videojs.EventTarget {
-  constructor(options) {
+  constructor(settings, options) {
```

Comment:
not necessary but maybe make `options = {}` optional. That way you don't need to add the empty objects in the test files. Maybe in `SyncController` as well